### PR TITLE
Fix CPP PreProcess error that regex match "if" in "ifndef"

### PR DESCRIPTION
**Pull request by ArdaFu**
Fix the AttributeError: 'NoneType' object has no attribute 'groups' when using SCons with Python 3.7.1. 

## Contributor Checklist:

* [x] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [X] I have updated `master/src/CHANGES.txt` directory (and read the `README.txt` in that directory)
* [X] I have updated the appropriate documentation

### CODE REVIEW

**Comment by bdbaddog** [comment]
Can you create a test to exercise the  bug?

**Comment by ArdaFu** [comment]
After tracing the code. I found that the bug is cpp regex match "if" in "ifndef".

**Comment by ArdaFu** [comment]
@bdbaddog New test case is added.

File: src/CHANGES.txt [hunk 1]
```diff
@@ -177,7 +177,8 @@ RELEASE 3.1.0.alpha.yyyymmdd - NEW DATE WILL BE INSERTED HERE
     - Typo in customized decider example in user guide
     - Replace usage of unittest.TestSuite with unittest.main() (fix #3113)
 
-
+  From Arda Fu
+    - Fix cpp regex error that match 'if' in 'ifndef'.
```
Comment (bdbaddog) [inline src/CHANGES.txt]: Change to be "Fix cpp scanner regex logic to treat ifndef. Previously it was not properly differentiating between if, ifdef, and ifndef"
Assuming this is accurate and I'm understanding this change..

**Pull request closed without merging**